File: SmartHR/SmartHR.Client/ClientApp/src/app/models/data/edit/grade-edit-model.ts
import { SalaryHeadModel } from "../salary-head-model";
import { SalaryStructureEditModel } from "./salary-structure-edit-model";

export interface GradeEditModel {
  gradeId?: number;
  gradeName?: string;
  basic?: number;
  salaryStructures?: SalaryStructureEditModel[]
}

File: SmartHR.DataApi/ViewModels/Edit/GradeEditModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace SmartHR.DataApi.ViewModels.Edit
{
    public class GradeEditModel
    {
        
        public int GradeId { get; set; }
        [Required, StringLength(30)]
        public string GradeName { get; set; }
        [Required]
        public decimal Basic { get; set; }
        public  SalaryStructureEditModel[] SalaryStructures { get; set; }
    }
}
